libuavcan/src/transport/uc_transfer_buffer.cpp

```diff
@@ -165,17 +190,40 @@ int TransferBufferManagerEntry::read(unsigned offset, uint8_t* data, unsigned le
     }
     UAVCAN_ASSERT((offset + len) <= max_write_pos_);
 
-    // This shall be optimized.
     unsigned total_offset = 0;
     unsigned left_to_read = len;
     uint8_t* outptr = data;
-    Block* p = blocks_.get();
+    Block* p;
+
+    // Check if this data starts in the previously cached block,
+    // if so, use it.
+    if (previous_block_ && (offset >= previous_block_start_offset_ && offset < (previous_block_start_offset_ + Block::Size)))
```

Comment:
The braces on the right side are placed awkwardly. Either the right side braces are not required or it's a bug.